src/Engine/Component/CameraComponents/CameraComponent.cpp

```diff
@@ -187,20 +145,67 @@ const glm::mat4 & CameraComponent::getProj() const {
     return m_projMat;
 }
 
-void CameraComponent::detUVW() {    
-    m_w = -Util::sphericalToCartesian(1.0f, m_theta, m_phi);
-    m_w = glm::vec3(m_w.x, m_w.z, -m_w.y); // one of the many reasons I like z to be up
-    m_v = Util::sphericalToCartesian(1.0f, m_theta, m_phi - glm::pi<float>() * 0.5f);
-    m_v = glm::vec3(m_v.x, m_v.z, -m_v.y);
-    m_u = glm::cross(m_v, m_w);
+void CameraComponent::detUVW() {
+    glm::vec3 w(-Util::sphericalToCartesian(1.0f, m_theta, m_phi));
+    w = glm::vec3(-w.y, w.z, -w.x); // one of the many reasons I like z to be up
+    glm::vec3 v(Util::sphericalToCartesian(1.0f, m_theta, m_phi - glm::pi<float>() * 0.5f));
+    v = glm::vec3(-v.y, v.z, -v.x);
+    glm::vec3 u(glm::cross(v, w));
+
+    // adjust relative to orientation of base
+    const glm::mat3 & orient(m_spatial->orientationMatrix());
+    setUVW(orient * u, orient * v, orient * w);
 }
 
 void CameraComponent::detView() const {
-    const glm::vec3 & pos(gameObject->getSpatial()->position());
-    m_viewMat = glm::lookAt(pos, pos - m_w, glm::vec3(0.0f, 1.0f, 0.0f));
-
+    m_viewMat = Util::viewMatrix(m_spatial->position(), u(), v(), w());
+    m_viewMatValid = true;
 }
 
 void CameraComponent::detProj() const {
-    m_projMat = glm::perspective(m_fov, Window::getAspectRatio(), m_near, m_far);
+    m_projMat = glm::perspective(m_fov, Window::getAspectRatio(), RenderSystem::near(), RenderSystem::far());
+    m_projMatValid = true;
+}
+
+void CameraComponent::detFrustum() const {
```

Comment:
@barlebino Will this work for an orthographic camera as well?

Reply:
I do not know, I cannot wrap my head around what getProj() * getView() does. However, I notice that detProj relies on m_fov which assumes that your view is shaped like a pyramid so it probably will not work with an orthographic camera. We can create a separate function for determining an orthographic frustum, and it should be easy to make one. The orthographic frustum function will likely require:

- Camera position (vec3f)
- Camera orientation (vec3f)
- Up vector (We can assume this is always [0, 1, 0])
- Width (float)
- Height (float)
- Near (float)
- Far (float)